src/activity_handlers.cpp

```diff
@@ -1596,9 +1597,10 @@ void activity_handlers::pulp_do_turn( player_activity *act, Character *you )
     for( item &corpse : corpse_pile ) {
         const mtype *corpse_mtype = corpse.get_mtype();
         if( !corpse.is_corpse() || !corpse.can_revive() ||
-            ( std::find( act->str_values.begin(), act->str_values.end(), "auto_pulp_no_acid" ) !=
-              act->str_values.end() && corpse_mtype->bloodType().obj().has_acid ) ) {
-            // Don't smash non-rezing corpses //don't smash acid zombies when auto pulping
+            ( ( std::find( act->str_values.begin(), act->str_values.end(), "auto_pulp_no_acid" ) !=
```

Comment:
you could pull out part of this into a variable e.g.;
`const bool acid_immune = you->is_immune_damage( damage_type::ACID ) && you->is_immune_field( fd_acid );`
then use it inside the if, the one below as well, and make this much more readable

json_flag_ACID_IMMUNE also seems unused in both files

Reply:
ah right. forgot to remove that bit. ill push out another commit

Reply:
no need for re-review you can just resolve the convo if you addressed the comments in one way or another (the person who starts it can't do it)